### PR TITLE
simplify build-main file

### PR DESCRIPTION
**Pull request by sayaliM0412**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
Simplify `build-main.yml` file and remove the use of custom logic which was copied from `build.yml`

### CODE REVIEW

File: .github/workflows/build-main.yml [hunk 1]
```diff
@@ -13,6 +13,25 @@ env:
...
+    runs-on: ubuntu-22.04
+    outputs:
+      timestamp: ${{ steps.get-date.outputs.date }}
+      setupSuccessful: "true"
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get current date
+        id: get-date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d %H:%M:%S')"
```
Comment (jnewton03) [inline .github/workflows/build-main.yml:27]: FYI set-output commands are being deprecated:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: .github/workflows/build-main.yml [hunk 2]
```diff
@@ -13,6 +13,25 @@ env:
...
+
+      - name: Get Latest Merge Commit SHA
+        id: get-sha
+        run: |
+          latest_merge_sha=$(git rev-parse origin/master)
+          echo "::set-output name=latestMergeSha::$latest_merge_sha"
```
Comment (jnewton03) [inline .github/workflows/build-main.yml:33]: FYI set-output commands are being deprecated:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

**Comment by sayaliM0412** [comment]
Workflow_dispatch run for this job->
https://github.com/liquibase/liquibase/actions/runs/5905305331
Functional Tests seems good as well ->
https://github.com/liquibase/liquibase-pro-tests/actions/runs/5905573773

File: .github/workflows/build-main.yml [hunk 3]
```diff
@@ -58,42 +65,10 @@ jobs:
 
       # Publish to GitHub Packages
       - name: Publish package
-        run: mvn -B clean deploy -pl '!liquibase-dist' -DskipTests=true "-Dbuild.repository.owner=${{ needs.setup.outputs.thisRepositoryOwner }}" "-Dbuild.repository.name=${{ needs.setup.outputs.thisRepositoryName }}" "-Dbuild.branch=${{ needs.setup.outputs.thisBranchName }}" "-Dbuild.number=${{ github.run_number }}" "-Dbuild.commit=${{ needs.setup.outputs.thisSha }}" "-Dbuild.timestamp=${{ needs.setup.outputs.timestamp }}"
+        run: mvn -B clean deploy -pl '!liquibase-dist' -DskipTests=true "-Dbuild.repository.owner=liquibase" "-Dbuild.repository.name=liquibase" "-Dbuild.branch=master" "-Dbuild.number=${{ github.run_number }}" "-Dbuild.commit=${{ needs.setup.outputs.latestMergeSha }}" "-Dbuild.timestamp=${{ needs.setup.outputs.timeStamp }}"
```
Comment (sayaliM0412) [inline .github/workflows/build-main.yml:68]: we could use `-DnewVersion="$(git rev-parse --short HEAD)-SNAPSHOT"` for sha-SNAPSHOT

**Comment by sayaliM0412** [comment]
Jira ticket : https://datical.atlassian.net/browse/DAT-15751

**Comment by jnewton03** [comment]
@sayaliM0412 are the failing integration tests expected here?

**Comment by sayaliM0412** [comment]
> @sayaliM0412 are the failing integration tests expected here?

Yes, I had raised a concern on #foundation-channel as well and they are working on a fix

**Comment by sayaliM0412** [comment]
> @sayaliM0412 would it be more efficient and/or simplify to combine the parallel jobs into a single sequential job? something like:
> 
> ```
> name: Build Main/Master SNAPSHOT
> 
> on:
>   workflow_dispatch:
>   schedule:
>     - cron: "5 6 * * *"
>   push:
>     branches:
>       - 'main'
>       - 'master'
> 
> env:
>   MAVEN_VERSION: '3.9.2'
> 
> jobs:
>   setup:
>     # ... [keep your setup job unchanged]
> 
>   build-and-publish:
>     name: Build & Publish
>     runs-on: ubuntu-22.04
>     needs: setup
>     permissions:
>       contents: read
>       packages: write
>     steps:
>       - uses: actions/checkout@v3
>       - name: Set up Java for publishing to GitHub Repository
>         uses: actions/setup-java@v3
>         with:
>           java-version: '17'
>           distribution: 'temurin'
>           cache: 'maven'
>           server-id: liquibase
> 
>       - name: Set up Maven
>         uses: stCarolas/setup-maven@v4.5
>         with:
>           maven-version: ${{ env.MAVEN_VERSION }}
> 
>       # Publish master-SNAPSHOT version
>       - name: Version Artifact as master-SNAPSHOT
>         run: mvn versions:set "-DnewVersion=master-SNAPSHOT"
>         
>       - name: Publish master-SNAPSHOT
>         run: mvn -B clean deploy -pl '!liquibase-dist' -DskipTests=true "-Dbuild.repository.owner=liquibase" "-Dbuild.repository.name=liquibase" "-Dbuild.branch=master" "-Dbuild.number=${{ github.run_number }}" "-Dbuild.commit=${{ needs.setup.outputs.latestMergeSha }}" "-Dbuild.timestamp=${{ needs.setup.outputs.timeStamp }}"
>         env:
>           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
> 
>       # Publish <commitsha>-SNAPSHOT version
>       - name: Version Artifact as commit-SNAPSHOT
>         run: mvn versions:set -DnewVersion="${{ needs.setup.outputs.latestMergeSha }}-SNAPSHOT"
>         
>       - name: Publish commit-SNAPSHOT
>         run: mvn -B clean deploy -pl '!liquibase-dist' -DskipTests=true "-Dbuild.repository.owner=liquibase" "-Dbuild.repository.name=liquibase" "-Dbuild.branch=master" "-Dbuild.number=${{ github.run_number }}" "-Dbuild.commit=${{ needs.setup.outputs.latestMergeSha }}" "-Dbuild.timestamp=${{ needs.setup.outputs.timeStamp }}"
>         env:
>           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
> ```

That looks better. I'll make that change. :)

**Pull request merged**